orbit_predictor/predictors/numerical.py

```diff
@@ -64,6 +65,54 @@ def sun_sync_plane_constellation(num_satellites, *,
         )
 
 
+def repeating_ground_track_sma(orbits, days=1, *, ecc, inc_deg=0, tolerance=1e-8):
+    """Computes semimajor axis for repeating ground track orbit.
+
+    Parameters
+    ----------
+    orbits : int
+        Number of orbits in a given period.
+    days : int, optional
+        Number of days to cover the given orbits, default to 1.
+    ecc : float
+        Eccentricity.
+    inc_deg : float, optional
+        Inclination in degrees, default to 0 (equatorial).
+
+    Returns
+    -------
+    sma : float
+        Semimajor axis.
+
+    Notes
+    -----
+    See Vallado "Fundamentals of Astrodynamics and Applications", 4th ed (2013)
+    and Wertz et al. "Space Mission Engineering: The New SMAD" (2011).
+
+    """
+    if not (isinstance(orbits, int) and isinstance(days, int)):
+        raise ValueError("Number of orbits and number of days must be integer.")
+
+    k = orbits / days
+    n = k * OMEGA_E
+
+    while True:
+        sma_new = np.cbrt(MU_E * (1 / n) ** 2)
+        p = sma_new * (1 - ecc ** 2)
+        Omega_dot = - 3 * n * J2 / 2 * (R_E_KM / p) ** 2 * np.cos(np.radians(inc_deg))
+        omega_dot = 3 * n * J2 / 4 * (R_E_KM / p) ** 2 * (4 - 5 * np.sin(np.radians(inc_deg)) ** 2)
+        M0_dot = (
```

Comment:
Looking at https://github.com/satellogic/orbit-predictor/files/3780018/repeat.pdf, I think there's a missing `n+` in this equation.

Reply:
Vallado uses `\dot{M}_0` whereas Eagle uses `\dot{M}`, which are related via `n` but I looked again at Vallado and this equation looks correct.